packages/nimble-components/src/table-column/text/group-header-view/index.ts

```diff
@@ -1,43 +1,32 @@
-import { observable, volatile } from '@microsoft/fast-element';
 import { DesignSystem } from '@microsoft/fast-foundation';
-import type { TableColumnWithPlaceholderColumnConfig } from '../../base/types';
-import { TableGroupHeaderView } from '../../base/group-header-view';
-import { template } from './template';
-import { styles } from './styles';
+import type { TableStringFieldValue } from '../../../table/types';
+import { TableColumnTextGroupHeaderViewBase } from '../../text-base/group-header-view';
+import { template } from '../../text-base/group-header-view/template';
+import { styles } from '../../text-base/group-header-view/styles';
+import type { TableColumnTextColumnConfig } from '..';
 
 declare global {
     interface HTMLElementTagNameMap {
         'nimble-table-column-text-group-header': TableColumnTextGroupHeaderView;
     }
 }
 /**
- * The custom element used to render a group row header for a group representing rows
- * grouped by a TableColumnText column.
+ * The group header view for displaying string fields as text.
  */
-export class TableColumnTextGroupHeaderView extends TableGroupHeaderView<
-string | null | undefined,
-TableColumnWithPlaceholderColumnConfig
+export class TableColumnTextGroupHeaderView extends TableColumnTextGroupHeaderViewBase<
+TableStringFieldValue,
+TableColumnTextColumnConfig
 > {
-    /** @internal */
-    public textSpan!: HTMLElement;
-
-    /** @internal */
-    @observable
-    public isValidContentAndHasOverflow = false;
-
-    @volatile
-    public get content(): string {
-        return typeof this.groupHeaderValue === 'string'
-            ? this.groupHeaderValue
-            : this.columnConfig?.placeholder ?? '';
+    public override get text(): string {
+        return this.groupHeaderValue!;
     }
 
-    public updateTitleOverflow(): void {
-        this.isValidContentAndHasOverflow = this.textSpan.offsetWidth < this.textSpan.scrollWidth;
+    public override get placeholder(): string {
+        return this.columnConfig?.placeholder ?? '';
     }
```

Comment:
Towards the other point, I think this one makes sense to not be volatile because the only observable property is the columnConfig and we know upstream that columnConfig is always replaced wholesale (maybe that assumption should be documented on the ColumnInternals.columnConfig property or something).

But if placeholder was an observable property and we didn't have that columnConfig assignment guarantee then this placeholder implementation would need to be marked `@volatile`

I dunno I think my hang-up is that it's pretty subtle and for an external client trying to extend could be easy to mess-up